demos/glacial_isostatic_adjustment/gia_adjoint_ice/gia_adjoint_ice.py

```diff
@@ -0,0 +1,717 @@
+radius_values = [6371e3, 6301e3, 5951e3, 5701e3, 3480e3]
+density_values = [3037, 3438, 3871, 4978]
+shear_modulus_values = [0.50605e11, 0.70363e11, 1.05490e11, 2.28340e11]
+viscosity_values = [2, -2, -2, -1.698970004]  # viscosity = 1e23 * 10**viscosity_values
+# N.b. that we have modified the viscosity of the Lithosphere viscosity from
+# Spada et al 2011 because we are using coarse grid resolution
+
+
+def initialise_background_field(field, background_values, vertical_tanh_width=40e3):
+    profile = background_values[0]
+    sharpness = 1 / vertical_tanh_width
+    depth = sqrt(X[0]**2 + X[1]**2)-radius_values[0]
+    for i in range(1, len(background_values)):
+        centre = radius_values[i] - radius_values[0]
+        mag = background_values[i] - background_values[i-1]
+        profile += step_func(depth, centre, mag, increasing=False, sharpness=sharpness)
+
+    field.interpolate(profile)
+
+
+density = Function(W, name="density")
+initialise_background_field(density, density_values)
+
+shear_modulus = Function(W, name="shear modulus")
+initialise_background_field(shear_modulus, shear_modulus_values)
+
+
+def bivariate_gaussian(x, y, mu_x, mu_y, sigma_x, sigma_y, rho, normalised_area=False):
+    arg = ((x-mu_x)/sigma_x)**2 - 2*rho*((x-mu_x)/sigma_x)*((y-mu_y)/sigma_y) + ((y-mu_y)/sigma_y)**2
+    numerator = exp(-1/(2*(1-rho**2))*arg)
+    if normalised_area:
+        denominator = 2*pi*sigma_x*sigma_y*(1-rho**2)**0.5
+    else:
+        denominator = 1
+    return numerator / denominator
+
+
+def setup_heterogenous_viscosity(viscosity):
+    heterogenous_viscosity_field = Function(viscosity.function_space(), name='viscosity')
+    antarctica_x, antarctica_y = -2e6, -5.5e6
+
+    low_viscosity_antarctica = bivariate_gaussian(X[0], X[1], antarctica_x, antarctica_y, 1.5e6, 0.5e6, -0.4)
+    heterogenous_viscosity_field.interpolate(-3*low_viscosity_antarctica + viscosity * (1-low_viscosity_antarctica))
+
+    llsvp1_x, llsvp1_y = 3.5e6, 0
+    llsvp1 = bivariate_gaussian(X[0], X[1], llsvp1_x, llsvp1_y, 0.75e6, 1e6, 0)
+    heterogenous_viscosity_field.interpolate(-3*llsvp1 + heterogenous_viscosity_field * (1-llsvp1))
+
+    llsvp2_x, llsvp2_y = -3.5e6, 0
+    llsvp2 = bivariate_gaussian(X[0], X[1], llsvp2_x, llsvp2_y, 0.75e6, 1e6, 0)
+    heterogenous_viscosity_field.interpolate(-3*llsvp2 + heterogenous_viscosity_field * (1-llsvp2))
+
+    slab_x, slab_y = 3e6, 4.5e6
+    slab = bivariate_gaussian(X[0], X[1], slab_x, slab_y, 0.7e6, 0.35e6, 0.7)
+    heterogenous_viscosity_field.interpolate(-1*slab + heterogenous_viscosity_field * (1-slab))
+
+    high_viscosity_craton_x, high_viscosity_craton_y = 0, 6.2e6
+    high_viscosity_craton = bivariate_gaussian(X[0], X[1], high_viscosity_craton_x, high_viscosity_craton_y, 1.5e6, 0.5e6, 0.2)
+    heterogenous_viscosity_field.interpolate(-1*high_viscosity_craton + heterogenous_viscosity_field * (1-high_viscosity_craton))
+
+    return heterogenous_viscosity_field
```

Comment:
I wonder if it'd be cleaner to just load these directly from the other demo (if they are indeed all the same), but then it's a decision of how to do the import from a sibling directory... `sys.path` modification? Make everything in `demos` part of the `gadopt` package so it can get a fully-qualified import?